test/Dialect/FIRRTL/dedup.mlir

```diff
@@ -445,6 +420,35 @@ firrtl.circuit "Flip" {
   }
 }
 
+// This is checking that the fixup phase due to changing bundle names does not
+// block the deduplication of parent modules.
+// CHECK-LABEL: firrtl.circuit "DelayedFixup"
+firrtl.circuit "DelayedFixup"  {
+  // CHECK: firrtl.extmodule @Foo
+  firrtl.extmodule @Foo(out a: !firrtl.bundle<a: uint<1>>)
+  // CHECK-NOT: firrtl.extmodule @Bar
+  firrtl.extmodule @Bar(out b: !firrtl.bundle<b: uint<1>>)
+  // CHECK: firrtl.module @Parent0
+  firrtl.module @Parent0(out %a: !firrtl.bundle<a: uint<1>>, out %b: !firrtl.bundle<b: uint<1>>) {
+    %foo_a = firrtl.instance foo  @Foo(out a: !firrtl.bundle<a: uint<1>>)
+    firrtl.connect %a, %foo_a : !firrtl.bundle<a: uint<1>>, !firrtl.bundle<a: uint<1>>
+    %bar_b = firrtl.instance bar  @Bar(out b: !firrtl.bundle<b: uint<1>>)
+    firrtl.connect %b, %bar_b : !firrtl.bundle<b: uint<1>>, !firrtl.bundle<b: uint<1>>
+  }
+  // CHECK-NOT: firrtl.module @Parent1
+  firrtl.module @Parent1(out %a: !firrtl.bundle<a: uint<1>>, out %b: !firrtl.bundle<b: uint<1>>) {
+    %foo_a = firrtl.instance foo  @Foo(out a: !firrtl.bundle<a: uint<1>>)
+    firrtl.connect %a, %foo_a : !firrtl.bundle<a: uint<1>>, !firrtl.bundle<a: uint<1>>
+    %bar_b = firrtl.instance bar  @Bar(out b: !firrtl.bundle<b: uint<1>>)
+    firrtl.connect %b, %bar_b : !firrtl.bundle<b: uint<1>>, !firrtl.bundle<b: uint<1>>
+  }
+  firrtl.module @DelayedFixup() {
+    // CHECK: firrtl.instance parent0  @Parent0
+    %parent0_a, %parent0_b = firrtl.instance parent0  @Parent0(out a: !firrtl.bundle<a: uint<1>>, out b: !firrtl.bundle<b: uint<1>>)
+    // CHECK: firrtl.instance parent1  @Parent0
+    %parent1_a, %parent1_b = firrtl.instance parent1  @Parent1(out a: !firrtl.bundle<a: uint<1>>, out b: !firrtl.bundle<b: uint<1>>)
+  }
```

Comment:
Should this test also check that the connects are wired up properly after dedup?

Reply:
I have dedicated tests for the connect verification + subfield ops.  This is really just testing that dedup is not "shooting itself in the foot".

Reply:
Sounds good to me. Thanks for keeping these tests lean. 💯